.github/workflows/linting.yml

```diff
@@ -15,6 +15,26 @@ jobs:
           python-version: '3.8'
           architecture: 'x64'
 
+      - name: Install Poetry
```

Comment:
The pre-commit check will keep failing even if the code is compliant because the pre-commit framework doesn't allow to run a poetry install before running a hook (i.e. pylint). Therefore we see the import errors and the action failed.
The pylint check should be moved to another job unrelated to pre-commit so that we can actually install poetry before lint the code.

Reply:
Yes, I agree.